weave/trait/power/rechargeable_battery_power_source_trait.proto

```diff
@@ -0,0 +1,475 @@
+/*
+ *
+ *    Copyright (c) 2019 Google LLC.
+ *    Copyright (c) 2016-2018 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file specifies a Weave Common trait that generalizes how
+ *      any Weave device resource presents dynamic, read-only battery power
+ *      source information common to any such device.
+ *
+ */
+
+syntax = "proto3";
+
+package weave.trait.power;
+
+import "wdl/wdl_options.proto";
+
+import "weave/common/time.proto";
+import "weave/trait/power/power_source_trait.proto";
+import "weave/trait/power/power_source_capabilities_trait.proto";
+import "weave/trait/power/battery_power_source_trait.proto";
+import "google/protobuf/wrappers.proto";
+
+option java_outer_classname = "WeaveInternalBatteryPowerSourceTrait";
+option objc_class_prefix = "SCM";
+
+/**
+ *  @brief
+ *    Rechargeable Battery Power Source
+ *
+ *  A Weave Common trait that allows vendors to enumerate dynamic, read-only
+ *  basic properties about a rechargeable battery power source in a device resource.
+ *
+ */
+message RechargeableBatteryPowerSourceTrait {
+        option (wdl.message_type) = TRAIT;
+
+        option (wdl.trait) = {
+          stability: ALPHA,
+                vendor_id: 0x0000,
+                id:        0x001D,
```

Comment:
It looks like this collides with ApplicationKeysTrait.

Reply:
How do I pick an ID? Is there a script for that or I just pick one and hope it works out? If so, it would be cool if the local "make check" worked on my machine...

Reply:
I had to just increment until it worked when I was trying to do this :/